localdev/run.sh

```diff
@@ -7,26 +7,31 @@ ARMADA_SVCS="armada-server lookout lookout-ingester executor binoculars jobservi
 cd "$(dirname "${0}")" || exit
 
 # start the kubernetes cluster if needed
-kind get clusters | grep armada-test &> /dev/null
-if [ $? -ne 0 ];
-then
+kind_output="$(kind get clusters)"
+if [[ $kind_output == *"armada-test"* ]]; then
+    echo "Using existing armada-test cluster"
+else
     scripts/kind-start.sh
 fi
 
 # select arm64 image for pulsar if needed
-uname -a | grep "arm64" &> /dev/null
-if [ $? -eq 0 ];
-then
-    export PULSAR_IMAGE="kezhenxu94/pulsar"
+os_system="$(uname -a)"
+#arm_found=$(grep -c arm64 <<< $os_system)
+#echo $arm_found
+if [[ $arm_found == *"arm64"* ]]; then
+    echo "Detecting Arm64 so will use custom pulsar image"
+    PULSAR_IMAGE="kezhenxu94/pulsar"
+else
+    echo "Detecting Linux so will use normal pulsar image."
 fi
 
 # see if pulsar is already up, in which case we don't need to sleep
-SLEEP_TIME=0
-docker-compose ps | grep -E "pulsar.+running" &> /dev/null
-if [ $? -ne 0 ];
+SLEEP_TIME=50
+compose_list="$(docker-compose ps)"
+if [[ $compose_list == *"pulsar.+running"* ]];
```

Comment:
does this regex pattern work for the bash matching?

Reply:
I think that is the wildcard part

Reply:
I see your point.  Is it a bad assumpition to just assume that if pulsar is running in docker environment it is running?